### PR TITLE
Adding support for loading YAML from a file.

### PR DESCRIPTION
**Pull request by divolgin**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.8
+FROM golang:1.11
```
Comment (divolgin) [inline Dockerfile]: This image no longer builds with go 1.8

File: cli/cmd/release_create.go [hunk 2]
```diff
@@ -7,68 +7,77 @@ import (
...
 
 func (r *runners) releaseCreate(cmd *cobra.Command, args []string) error {
-	if createReleaseYaml == "" {
+	if r.args.createReleaseYaml == "" && r.args.createReleaseYamlFile == "" {
 		return fmt.Errorf("yaml is required")
 	}
 
-	if createReleaseYaml == "-" {
+	if r.args.createReleaseYaml != "" && r.args.createReleaseYamlFile != "" {
+		return fmt.Errorf("only yaml or yaml-file has to be specified")
```
Comment (dexhorthy) [inline cli/cmd/release_create.go]: mind if we re-word this a little bit?
```suggestion
		return fmt.Errorf("only one of yaml or yaml-file may be specified")
```

File: cli/cmd/root.go [hunk 3]
```diff
@@ -77,35 +80,62 @@ Use "{{.CommandPath}} [command] --help" for more information about a command.{{e
...
+	runCmds.InitReleaseInspect(releaseCmd)
+	runCmds.IniReleaseList(releaseCmd)
+	runCmds.InitReleaseUpdate(releaseCmd)
+	runCmds.InitReleasePromote(releaseCmd)
+	runCmds.InitReleaseLint(releaseCmd)
+
+	runCmds.rootCmd.AddCommand(shipReleaseCommand)
+	runCmds.InitShipReleaseCommand(shipReleaseCommand)
+
+	runCmds.rootCmd.SetUsageTemplate(rootCmdUsageTmpl)
```
Comment (dexhorthy) [inline cli/cmd/root.go]: ❤️

**Comment by dexhorthy** [comment]
One comment on the error wording, otherwise this looks like a really solid refactor. What have you done to test / plans for testing?

**Pull request merged**